### ISSUE TITLE
turn a function JdbcConnection => R into a DBIOAction

### ISSUE THREAD

**Issue by cvogt** [slick/slick#1579]
I am proposing the addition of an unsafe DBIOAction, e.g.

```
DBIOAction.unsafeJdbc{
  jdbcConnection => /* do whatever */
}
```

that allows integration of things build on jdbc sessions with slick actions. This would solve this: https://github.com/slick/slick/issues/1574 and could be a building block in solving this: https://github.com/slick/slick/issues/1558 

More generally this is a way to break out of slick's controlled model into an unsafe world without restrictions and free interop with any JDBC-based things Slick may not support directly.

**Comment by federico-pellegatta** [comment]
What about `SimpleDBIO` @ [StreamingInvokerAction.scala#L69](https://github.com/slick/slick/blob/master/slick/src/main/scala/slick/jdbc/StreamingInvokerAction.scala#L69). Am I missing something?

**Comment by nafg** [comment]
Yes that exists. But certainly a nicer API (analogous to DBIO.successful
vs. SuccessAction, also the function could take a Connection rather than a
Context perhaps) would be nice. (That was a tautology...)

Conversely it would be nice to have an UnmanagedDatabase, so that if I am
handed a Connection I can run Actions (e.g., to implement a Flyway Java
migration). You can see a hackish implementation here:
https://github.com/101dev/slick-migration-api-flyway/blob/master/src/main/scala/scala/slick/migration/flyway/UnmanagedDatabase.scala
